src/main/java/io/confluent/copycat/hdfs/HdfsSinkTask.java

```diff
@@ -43,37 +43,51 @@ public HdfsSinkTask() {
   public void start(Properties props) {
     try {
       HdfsSinkConnectorConfig connectorConfig = new HdfsSinkConnectorConfig(props);
-      hdfsWriter = new HdfsWriter(connectorConfig);
-      context.resetOffset(hdfsWriter.getPreviousOffsets());
+      assignment = context.assignment();
+      hdfsWriter = new HdfsWriter(connectorConfig, assignment);
+      recover(assignment);
+      resetOffsets();
     } catch (ConfigException e) {
-      throw new CopycatRuntimeException(
+      throw new CopycatException(
           "Couldn't start HdfsSinkConnector due to configuration error.", e);
     }
+
   }
 
   @Override
   public void stop() throws CopycatException {
-    try {
-      hdfsWriter.close();
-    } catch (IOException e) {
-      throw new CopycatRuntimeException("Error closing the HDFS writer", e);
-    }
+    hdfsWriter.close();
+
   }
 
   @Override
   public void put(Collection<SinkRecord> records) throws CopycatException {
     String topic;
     int partition;
-    for (SinkRecord record: records) {
-      topic = record.getTopic();
-      partition = record.getPartition();
+    for (SinkRecord record : records) {
+      topic = record.topic();
+      partition = record.kafkaPartition();
       TopicPartition topicPart = new TopicPartition(topic, partition);
-      hdfsWriter.writeRecord(topicPart, record);
+      if (assignment.contains(topicPart)) {
+        hdfsWriter.writeRecord(topicPart, record);
+      } else {
+        log.warn("Seen record {} from not assigned TopicPartition {}", record, topicPart);
+      }
     }
   }
 
   @Override
-  public void flush(Map<TopicPartition, Long> offsets) {
-    // Do nothing as we manage the offset
+  public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) {
+    // Do nothing as the connector manages the offset
```

Comment:
I know we can safely ignore the `flush` request, but should we always ignore it? We can probably leave this for later, but are there cases where we'd want to commit at this point just to avoid wasting the effort we've put in so far to write data up to this point?